### PR TITLE
Starman DX

### PR DESCRIPTION
**Pull request by ghost**
This finishes all the stuff I didn't get to finish (or got too lazy to finish before).
Adds psi lifeup alpha, 25 sec cd to heal 30 of each damage type and your components.
The starman becomes significantly tankier. Bullets, especially weaker ones, have a high chance to deflect off the starman's armor. Weak brute items like a wrench do nothing at all. Lasers do a decent amount, but the base armor and absorb have been raised as well. Keep in mind the starman is literally incapable of moving, so some armor is needed.
Added a pulse rifle named psi beam to the starman, 4 sec CD on each fire.
Added a hailer that plays some spooky earthbound sound effects.
Improved naming scheme for starmen.
Added precast psi sound that matches Earthbound's.

All in all this will make the starman quite the formidable foe, perfect for mid-high threat adminbussing. 

:cl:
 * rscadd: Starman bus is now fully functional.

### CODE REVIEW

**Comment by ghost** [comment]
Fucks sake I didn't bother getting latest version before doing this. Will fix in the morning.

**Comment by ghost** [comment]
Also fucked up the commit
I feel Shifty's fingers reaching towards his keyboard, ready to type two words: enterprise and grade.

**Comment by DrSnips** [comment]
A pulse rifle shot id a bit much don't you think?

**Comment by ghost** [comment]
It's got a decent CD and after testing it out vs some mobs the inability to move really hinders you, so this helps it stay quite dangerous.

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -639,12 +639,13 @@
 		"Basic" = "starman",
 	)
 	speed_modifier = CYBORG_STARMAN_SPEED_MODIFIER
+	default_modules = 0
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: null. 0 isn't a list.

File: code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm [hunk 2]
```diff
@@ -215,4 +297,41 @@
...
+	
+/obj/item/weapon/gun/energy/starman_beam
+	name = "Psi Beam Vestibule"
+	desc = "Used to channel psionic energy into a deadly form."
+	icon_state = "pulse"
+	item_state = "gun"
+	fire_sound = 'sound/effects/psi/psi_beam.ogg'
+	fire_delay = 3 SECONDS
+	cell_type = "/obj/item/weapon/cell/infinite"
+	projectile_type = "/obj/item/projectile/beam/pulse"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm:311]: jesus christ

File: code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm [hunk 3]
```diff
@@ -215,4 +297,41 @@
...
+	
+	
+/obj/item/weapon/gun/energy/starman_beam
+	name = "Psi Beam Vestibule"
+	desc = "Used to channel psionic energy into a deadly form."
+	icon_state = "pulse"
+	item_state = "gun"
+	fire_sound = 'sound/effects/psi/psi_beam.ogg'
+	fire_delay = 3 SECONDS
+	cell_type = "/obj/item/weapon/cell/infinite"
```
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm:310]: now we're getting wonky levels that shouldn't be possible

**Comment by SonixApache** [comment]
🤔

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 1 above)
Comment (ghost) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: I'm 99% sure this is a boolean that dictates whether the flash and flashlight are added to robot stuff.

File: code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm (hunk 2 above)
Comment (ghost) [reply code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm:311]: Again, a reliable way to shoot shit is needed when you're an immobile rock. It's a medium-high danger adminbus, so keep that in mind.

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 1 above)
Comment (jknpj) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: I am mixing bleeding edge with my WiP branch. Att least change this to FALSE for consistency.

**Comment by ghost** [comment]
Just merge it ya wanker, it's adminbus

**Comment by kilozombie** [comment]
It's adminbus but didn't you say once upon a time that you wanted it to be a possible event or even antagonist for dynamic?

**Comment by ghost** [comment]
Maybe in the distant future, but I do genuinely believe it is quite weak without a reliable, heavy hitting weapon. Two people unloading laser rifles on it will make quick work of it unless the starman is able to quickly deploy its psi wall or mindshock them.

**Comment by jknpj** [comment]
isn't that the point

**Comment by ghost** [comment]
Yeah, so the pulse rifle is balanced

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 1 above)
Comment (jknpj) [inline code/modules/mob/living/silicon/robot/robot_modules.dm]: No, really, at least change it to FALSE

Comment (ghost) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: var/default_modules = TRUE //Do we start with a flash/light?

I could make it FALSE for clarity but I would hope a programmer would know that 0 is false...

Comment (jknpj) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: I changed every borg binary check to use TRUE/FALSE. Keep it consistent.

Comment (ghost) [reply code/modules/mob/living/silicon/robot/robot_modules.dm]: Alright BY*ND coder

File: code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm [hunk 4]
```diff
@@ -11,14 +11,58 @@
...
-	set_module_sprites(list("Basic" = "starman"))
 	src.add_spell(new /spell/aoe_turf/starman_play_music)
 	src.add_spell(new /spell/aoe_turf/starman_brainshock)
 	src.add_spell(new /spell/aoe_turf/starman_starstorm)
+	src.add_spell(new /spell/aoe_turf/starman_heal)
 	src.add_spell(new /spell/targeted/starman_shield)
 	src.add_spell(new /spell/targeted/starman_warp)
 
-			
+/mob/living/silicon/robot/updatename(var/prefix)
```
Comment (SonixApache) [inline code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm:21]: good job

Comment (jknpj) [reply code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm:21]: god fucking damn it

Comment (jknpj) [reply code/modules/mob/living/silicon/robot/robot_subtypes/starman.dm:21]: @GlassEclipse fix it fix it fix it

**Pull request merged**